storage/series/series.go

```diff
@@ -57,9 +57,14 @@ var (
 )
 
 var (
-	nilID = ts.BinaryID(checked.NewBytes(nil, nil))
+	nilIDBytes = checked.NewBytes(nil, nil)
+	nilID      = ts.BinaryID(nilIDBytes)
 )
 
+func init() {
+	nilIDBytes.IncRef()
```

Comment:
I'm not sure about the package initialization ordering, but potentially `IncRef()` will be called after `ts.BinaryID()`, could you verify it's not what's happening?